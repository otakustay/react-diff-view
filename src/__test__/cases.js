/* In case you are looking for type of hunk, see src/Hunk/index.js */
import {parseDiff} from '..';

export const basic = `diff --git a/src/__test__/index.test.jsx b/src/__test__/index.test.jsx
index 643c2f0..7883597 100644
--- a/src/__test__/index.test.jsx
+++ b/src/__test__/index.test.jsx
@@ -21,3 +21,3 @@ describe('basic test', () => {
     test('App renders correctly', () => {
-        expect(renderer.create(<App diffText={'deff'} />).toJSON()).toMatchSnapshot();
+        expect(renderer.create(<App diffText={'diff'} />).toJSON()).toMatchSnapshot();
     });`;

export const basicHunk = parseDiff(basic)[0].hunks[0];

export const multiple = `diff --git a/src/index1.js b/src/index1.js
index e69de29..d00491f 100644
--- a/src/index1.js
+++ b/src/index1.js
@@ -0,0 +1 @@
+1
diff --git a/src/index.js b/src/index2.js
index d00491f..0cfbf08 100644
--- a/src/index2.js
+++ b/src/index2.js
@@ -1 +1 @@
-1
+2
diff --git a/src/index3.js b/src/index3.js
index 0cfbf08..e69de29 100644
--- a/src/index3.js
+++ b/src/index3.js
@@ -1 +0,0 @@
-2`;

export const multipleHunk = parseDiff(basic)[0].hunks[0];
