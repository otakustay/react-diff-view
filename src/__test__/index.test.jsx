import renderer from 'react-test-renderer';
import {parseDiff, Diff, Hunk} from '..';
import {basic, multiple} from './cases';

const App = ({diffText, viewType = 'split'}) => {
    const files = parseDiff(diffText);

    const renderFile = ({oldRevision, newRevision, type, hunks}) => (
        <Diff key={oldRevision + '-' + newRevision} viewType={viewType} diffType={type} hunks={hunks}>
            {hunks => hunks.map(hunk => <Hunk key={hunk.content} hunk={hunk} />)}
        </Diff>
    );

    return (
        <div>
            {files.map(renderFile)}
        </div>
    );
};

describe('App with viewType === split', () => {
    test('renders correctly', () => {
        expect(renderer.create(<App diffText={basic} />).toJSON()).toMatchSnapshot();
    });

    test('multiple diff', () => {
        expect(renderer.create(<App diffText={multiple} />).toJSON()).toMatchSnapshot();
    });
});

describe('App with viewType === unified', () => {
    test('renders correctly', () => {
        expect(renderer.create(<App diffText={basic} viewType="unified" />).toJSON()).toMatchSnapshot();
    });

    test('multiple diff', () => {
        expect(renderer.create(<App diffText={multiple} viewType="unified" />).toJSON()).toMatchSnapshot();
    });
});
