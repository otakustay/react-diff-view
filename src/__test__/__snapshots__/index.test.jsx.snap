// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App with viewType === split multiple diff 1`] = `
<div>
  <table
    className="diff diff-split"
    onMouseDown={[Function]}
  >
    <colgroup>
      <col
        className="diff-gutter-col"
      />
      <col />
      <col
        className="diff-gutter-col"
      />
      <col />
    </colgroup>
    <tbody
      className="diff-hunk"
    >
      <tr
        className="diff-line diff-line-new-only"
      >
        <td
          className="diff-gutter diff-gutter-omit"
        />
        <td
          className="diff-code diff-code-omit"
        />
        <td
          className="diff-gutter diff-gutter-insert"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          1
        </td>
        <td
          className="diff-code diff-code-insert"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          1
        </td>
      </tr>
    </tbody>
  </table>
  <table
    className="diff diff-split"
    onMouseDown={[Function]}
  >
    <colgroup>
      <col
        className="diff-gutter-col"
      />
      <col />
      <col
        className="diff-gutter-col"
      />
      <col />
    </colgroup>
    <tbody
      className="diff-hunk"
    >
      <tr
        className="diff-line diff-line-compare"
      >
        <td
          className="diff-gutter diff-gutter-delete"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          1
        </td>
        <td
          className="diff-code diff-code-delete"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          1
        </td>
        <td
          className="diff-gutter diff-gutter-insert"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          1
        </td>
        <td
          className="diff-code diff-code-insert"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          2
        </td>
      </tr>
    </tbody>
  </table>
  <table
    className="diff diff-split"
    onMouseDown={[Function]}
  >
    <colgroup>
      <col
        className="diff-gutter-col"
      />
      <col />
      <col
        className="diff-gutter-col"
      />
      <col />
    </colgroup>
    <tbody
      className="diff-hunk"
    >
      <tr
        className="diff-line diff-line-old-only"
      >
        <td
          className="diff-gutter diff-gutter-delete"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          1
        </td>
        <td
          className="diff-code diff-code-delete"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          2
        </td>
        <td
          className="diff-gutter diff-gutter-omit"
        />
        <td
          className="diff-code diff-code-omit"
        />
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`App with viewType === split renders correctly 1`] = `
<div>
  <table
    className="diff diff-split"
    onMouseDown={[Function]}
  >
    <colgroup>
      <col
        className="diff-gutter-col"
      />
      <col />
      <col
        className="diff-gutter-col"
      />
      <col />
    </colgroup>
    <tbody
      className="diff-hunk"
    >
      <tr
        className="diff-line diff-line-normal"
      >
        <td
          className="diff-gutter diff-gutter-normal"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          21
        </td>
        <td
          className="diff-code diff-code-normal"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
              test('App renders correctly', () =&gt; {
        </td>
        <td
          className="diff-gutter diff-gutter-normal"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          21
        </td>
        <td
          className="diff-code diff-code-normal"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
              test('App renders correctly', () =&gt; {
        </td>
      </tr>
      <tr
        className="diff-line diff-line-compare"
      >
        <td
          className="diff-gutter diff-gutter-delete"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          22
        </td>
        <td
          className="diff-code diff-code-delete"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
                  expect(renderer.create(&lt;App diffText={'deff'} /&gt;).toJSON()).toMatchSnapshot();
        </td>
        <td
          className="diff-gutter diff-gutter-insert"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          22
        </td>
        <td
          className="diff-code diff-code-insert"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
                  expect(renderer.create(&lt;App diffText={'diff'} /&gt;).toJSON()).toMatchSnapshot();
        </td>
      </tr>
      <tr
        className="diff-line diff-line-normal"
      >
        <td
          className="diff-gutter diff-gutter-normal"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          23
        </td>
        <td
          className="diff-code diff-code-normal"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
              });
        </td>
        <td
          className="diff-gutter diff-gutter-normal"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          23
        </td>
        <td
          className="diff-code diff-code-normal"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
              });
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`App with viewType === unified multiple diff 1`] = `
<div>
  <table
    className="diff diff-unified"
    onMouseDown={[Function]}
  >
    <colgroup>
      <col
        className="diff-gutter-col"
      />
      <col
        className="diff-gutter-col"
      />
      <col />
    </colgroup>
    <tbody
      className="diff-hunk"
    >
      <tr
        className="diff-line"
      >
        <td
          className="diff-gutter diff-gutter-insert"
        />
        <td
          className="diff-gutter diff-gutter-insert"
        >
          1
        </td>
        <td
          className="diff-code diff-code-insert"
        >
          1
        </td>
      </tr>
    </tbody>
  </table>
  <table
    className="diff diff-unified"
    onMouseDown={[Function]}
  >
    <colgroup>
      <col
        className="diff-gutter-col"
      />
      <col
        className="diff-gutter-col"
      />
      <col />
    </colgroup>
    <tbody
      className="diff-hunk"
    >
      <tr
        className="diff-line"
      >
        <td
          className="diff-gutter diff-gutter-delete"
        >
          1
        </td>
        <td
          className="diff-gutter diff-gutter-delete"
        />
        <td
          className="diff-code diff-code-delete"
        >
          1
        </td>
      </tr>
      <tr
        className="diff-line"
      >
        <td
          className="diff-gutter diff-gutter-insert"
        />
        <td
          className="diff-gutter diff-gutter-insert"
        >
          1
        </td>
        <td
          className="diff-code diff-code-insert"
        >
          2
        </td>
      </tr>
    </tbody>
  </table>
  <table
    className="diff diff-unified"
    onMouseDown={[Function]}
  >
    <colgroup>
      <col
        className="diff-gutter-col"
      />
      <col
        className="diff-gutter-col"
      />
      <col />
    </colgroup>
    <tbody
      className="diff-hunk"
    >
      <tr
        className="diff-line"
      >
        <td
          className="diff-gutter diff-gutter-delete"
        >
          1
        </td>
        <td
          className="diff-gutter diff-gutter-delete"
        />
        <td
          className="diff-code diff-code-delete"
        >
          2
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`App with viewType === unified renders correctly 1`] = `
<div>
  <table
    className="diff diff-unified"
    onMouseDown={[Function]}
  >
    <colgroup>
      <col
        className="diff-gutter-col"
      />
      <col
        className="diff-gutter-col"
      />
      <col />
    </colgroup>
    <tbody
      className="diff-hunk"
    >
      <tr
        className="diff-line"
      >
        <td
          className="diff-gutter diff-gutter-normal"
        >
          21
        </td>
        <td
          className="diff-gutter diff-gutter-normal"
        >
          21
        </td>
        <td
          className="diff-code diff-code-normal"
        >
              test('App renders correctly', () =&gt; {
        </td>
      </tr>
      <tr
        className="diff-line"
      >
        <td
          className="diff-gutter diff-gutter-delete"
        >
          22
        </td>
        <td
          className="diff-gutter diff-gutter-delete"
        />
        <td
          className="diff-code diff-code-delete"
        >
                  expect(renderer.create(&lt;App diffText={'deff'} /&gt;).toJSON()).toMatchSnapshot();
        </td>
      </tr>
      <tr
        className="diff-line"
      >
        <td
          className="diff-gutter diff-gutter-insert"
        />
        <td
          className="diff-gutter diff-gutter-insert"
        >
          22
        </td>
        <td
          className="diff-code diff-code-insert"
        >
                  expect(renderer.create(&lt;App diffText={'diff'} /&gt;).toJSON()).toMatchSnapshot();
        </td>
      </tr>
      <tr
        className="diff-line"
      >
        <td
          className="diff-gutter diff-gutter-normal"
        >
          23
        </td>
        <td
          className="diff-gutter diff-gutter-normal"
        >
          23
        </td>
        <td
          className="diff-code diff-code-normal"
        >
              });
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
